101-recovery-services-create-vm-and-configure-backup/azuredeploy.json

```diff
@@ -0,0 +1,225 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "adminUsername": {
+      "type": "string",
+      "metadata": {
+        "description": "Username for the Virtual Machine."
+      }
+    },
+    "adminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Password for the Virtual Machine."
+      }
+    },
+    "dnsLabelPrefix": {
+      "type": "string",
+      "metadata": {
+        "description": "Unique DNS Name for the Public IP used to access the Virtual Machine."
+      }
+    },
+    "windowsOSVersion": {
+      "type": "string",
+      "defaultValue": "2012-R2-Datacenter",
+      "allowedValues": [
+        "2008-R2-SP1",
+        "2012-Datacenter",
+        "2012-R2-Datacenter"
+      ],
+      "metadata": {
+        "description": "The Windows version for the VM. This will pick a fully patched image of this given Windows version. Allowed values: 2008-R2-SP1, 2012-Datacenter, 2012-R2-Datacenter."
+      }
+    },
+    "vaultName": {

```

Comment:
does this need to be globally unique?  If not, just move it to a variable

Reply:
Unique in subscription. We want user to give vault name as parameter, and hence I want to keep it parameter. Any pattern when to use variables vs parameter?

Reply:
Updated as per review comment. But just wanted to know when to use variable/parameter?

Reply:
Generally if it's an endpoint or the user needs to know it for some reason, parameter is good - and even then a defaultValue is helpful (generate a unique one)